# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineCacheIRCompiler.h
# Commit: 9882c0d924b8
# Full Hash: 9882c0d924b8ad13e1e0828f9e3860244ad6f33f
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2023-07-08 09:31:48
# Regressor Bug: 1837620
# File Overlap Count: 1
# Description:
#   Bug 1837620 - Part 6: Make edges for multiple shape guard weak too r=sfink
#   
#   To make the edges for the multiple shape guard weak we need a way to find and
#   update the objects used to hold these edges. The simplest way to do this is to
#   add a vector in the zone.  I don't believe that there are not huge numbers of
# ==============================================================================

diff -r ba12c0cbb7a4 -r 9882c0d924b8 js/src/jit/BaselineCacheIRCompiler.h
--- a/js/src/jit/BaselineCacheIRCompiler.h	Fri Jul 07 17:05:44 2023 +0000
+++ b/js/src/jit/BaselineCacheIRCompiler.h	Fri Jul 07 17:05:44 2023 +0000
@@ -144,6 +144,19 @@
   CACHE_IR_COMPILER_UNSHARED_GENERATED
 };
 
+// Special object used for storing a list of shapes to guard against. These are
+// only used in the fields of CacheIR stubs and do not escape.
+class ShapeListObject : public ListObject {
+ public:
+  static const JSClass class_;
+  static const JSClassOps classOps_;
+  static ShapeListObject* create(JSContext* cx);
+  static void trace(JSTracer* trc, JSObject* obj);
+
+  Shape* get(uint32_t index);
+  bool traceWeak(JSTracer* trc);
+};
+
 }  // namespace jit
 }  // namespace js
 